syft/get_source.go

```diff
@@ -0,0 +1,102 @@
+package syft
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/anchore/syft/syft/source"
+)
+
+func GetSource(ctx context.Context, userInput string, getSourceConfig ...GetSourceConfig) (source.Source, error) {
+	cfg := DefaultGetSourceConfig()
+	if len(getSourceConfig) > 1 {
+		return nil, fmt.Errorf("the optional GetSourceConfig requires a single value; got multiple: %v", getSourceConfig)
+	}
+	if len(getSourceConfig) == 1 {
+		cfg = getSourceConfig[0]
+	}
+
+	// must set userInput before getting source providers, it is important configuration for them all
+	cfg.SourceProviderConfig.UserInput = userInput
+
+	providers, err := cfg.GetProviders()
+	if err != nil {
+		return nil, err
+	}
+
+	var errs []error
+	var fileNotfound error
+
+	// call each source provider until we find a valid source
+	for _, p := range providers {
+		src, err := p.Provide(ctx)
+		if err != nil {
+			err = eachError(err, func(err error) error {
+				if errors.Is(err, os.ErrNotExist) {
+					fileNotfound = err
+					return nil
+				}
+				return err
+			})
+			if err != nil {
+				errs = append(errs, err)
+			}
+		}
+		if src != nil {
+			// if we have a non-image type and platform is specified, it's an error
+			// if _, ok := src.(*stereoscope.StereoscopeImageSource); !ok && cfg.Platform != nil {
+			//	return src, fmt.Errorf("invalid argument: --platform specified with non-image source")
+			//}
+			return src, nil
```

Comment:
the comment doesn't match the code commented out... is this dead? or in need of refactor?

Reply:
comment was accurate (note the `!ok`); restored this check